app/views/participants/index.haml

```diff
@@ -31,11 +36,17 @@
       %tr
         %td= f.text_field :name, placeholder: 'Name', class: 'form-control'
         %td= f.number_field :number, class: 'form-control'
+        %td= f.collection_select :bus_id, Bus.all, :id, :number,
+          {include_blank: 'Bus number'}, {class: 'form-control'}
         %td= f.submit 'Add', class: 'btn btn-primary'
     - @participants.each do |participant|
       %tr
         = form_for participant do |f|
-          %td= f.text_field :name, value: participant.name, class: 'form-control'
-          %td= f.number_field :number, value: participant.number, class: 'form-control'
+          %td= f.text_field :name, value: participant.name,
+            class: 'form-control', required: true
+          %td= f.number_field :number, value: participant.number,
+            class: 'form-control'
+          %td= f.collection_select :bus_id, Bus.all, :id, :number,
```

Comment:
Better to:

1. serve this as an instance variable

2. make sure they're ordered by number, either explicitly or with a `default_scope`